robot/web/web.go

```diff
@@ -481,8 +481,18 @@ func (svc *webService) updateResources(resources map[resource.Name]resource.Reso
 			if ok {
 				apiResColl = reg.MakeEmptyCollection()
 			} else {
-				svc.logger.Debugw("making heterogeneous API resource collection", "api", a)
-				apiResColl = resource.NewEmptyAPIResourceCollection[resource.Resource](a)
+				// log a warning here to remind users to register their APIs, but can ignore if client communication is not wanted/needed
+				// or if resource is internal to RDK
+				if a.Type.Namespace != resource.APINamespaceRDKInternal {
+					for n := range v {
+						if !n.ContainsRemoteNames() {
+							svc.logger.Warnw(
+								"missing registration for api, resources with this API will be unreachable through a client", "api", n.API)
```

Comment:
This isn't true though, is it? That's the whole point of foreign resources. Only the client and module need to have the registration. The parent robot doesn't need it at all. Or am I misunderstanding the resource(s) that this is being applied to?

Reply:
this is applied only for local resources (so the warning should not appear for foreign resources since they should always be remote)